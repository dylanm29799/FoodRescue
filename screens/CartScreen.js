import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";
import { scale } from "../components/ResponsiveText";
import Colour from "../constants/Colour";

const CartScreen = (props) => {
  //GettingParams
  const title = props.navigation.getParam("Title");
  const quantity = props.navigation.getParam("quantity");
  const initialPrice = props.navigation.getParam("initialPrice");
  const price = props.navigation.getParam("price");

  const newPrice = price * quantity;
  const newInitial = initialPrice * quantity;
  const savings = newInitial - newPrice;
  const newSavings = Math.round(savings * 100) / 100;
  return (
    <View style={styles.screen}>
      <View style={{ borderWidth: 1, width: "80%" }}>
        <View style={{}}>
          <Text style={styles.title}>You are Buying: </Text>

          <Text style={styles.title}>{title}</Text>
        </View>
        <Text style={styles.quantity}>Qunatity: {quantity}</Text>
        <Text style={styles.price}>Total Price: €{newPrice}</Text>
        <Text style={styles.initialPrice}>You are saving: €{newSavings}</Text>
      </View>
      <View style={{ flexDirection: "row" }}>
        <TouchableOpacity
          onPress={() => {
            props.navigation.navigate({
              routeName: "Cart",
            });
          }}
          style={styles.touchable}
        >
          <Text style={styles.touchableText}>Pay With Cash</Text>
        </TouchableOpacity>

        <TouchableOpacity
          onPress={() => {
            props.navigation.navigate({
              routeName: "Cart",
            });
          }}
          style={styles.touchable}
        >
          <Text style={styles.touchableText}>Pay With Card</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },

  title: {
    textAlign: "center",
    fontSize: scale(20),
    marginTop: scale(10),
  },
  quantity: {
    textAlign: "center",
    fontSize: scale(20),
    marginTop: scale(10),
  },
  initialPrice: {
    textAlign: "center",
    fontSize: scale(20),
    marginTop: scale(10),
  },
  price: {
    textAlign: "center",
    fontSize: scale(20),
    marginTop: scale(10),
  },
  touchable: {
    height: scale(30),
    width: scale(70),
    backgroundColor: Colour.primaryColour,
    justifyContent: "center",
    marginTop: scale(30),
    marginHorizontal: scale(10),
  },
  touchableText: {
    color: "#fff",
    textAlign: "center",
    fontSize: scale(12),
  },
});

export default CartScreen;
