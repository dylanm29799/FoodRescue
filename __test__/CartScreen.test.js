// __tests__/CartScreen.test.js

import React from "react";
import CartScreen from "../screens/CartScreen";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";
firebase.initializeApp(firebaseConfig);
test("renders correctly", () => {
  const tree = renderer.create(<CartScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
