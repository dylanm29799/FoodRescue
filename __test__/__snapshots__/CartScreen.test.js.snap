// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "borderWidth": 1,
        "width": "80%",
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        style={
          Object {
            "fontSize": 42.857142857142854,
            "marginTop": 21.428571428571427,
            "textAlign": "center",
          }
        }
      >
        You are Buying: 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 42.857142857142854,
            "marginTop": 21.428571428571427,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 42.857142857142854,
          "marginTop": 21.428571428571427,
          "textAlign": "center",
        }
      }
    >
      Qunatity: 
      
    </Text>
    <Text
      style={
        Object {
          "fontSize": 42.857142857142854,
          "marginTop": 21.428571428571427,
          "textAlign": "center",
        }
      }
    >
      Total Price: €
      0
    </Text>
    <Text
      style={
        Object {
          "fontSize": 42.857142857142854,
          "marginTop": 21.428571428571427,
          "textAlign": "center",
        }
      }
    >
      You are saving: €
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#c44536",
          "height": 64.28571428571428,
          "justifyContent": "center",
          "marginHorizontal": 21.428571428571427,
          "marginTop": 64.28571428571428,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 25.714285714285715,
            "textAlign": "center",
          }
        }
      >
        Pay With Cash
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#c44536",
          "height": 64.28571428571428,
          "justifyContent": "center",
          "marginHorizontal": 21.428571428571427,
          "marginTop": 64.28571428571428,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 25.714285714285715,
            "textAlign": "center",
          }
        }
      >
        Pay With Card
      </Text>
    </View>
  </View>
</View>
`;
